### ISSUE TITLE
Consider running without syscall tracepoints

### ISSUE THREAD

**Issue by umanwizard** [open-telemetry/opentelemetry-ebpf-profiler#173]
This discussion began in Slack: https://cloud-native.slack.com/archives/C03J794L0BV/p1728340374847259

The issue is that the agent refuses to run when syscall tracepoints are disabled; that is, when the path `/sys/kernel/debug/tracing/events/syscalls` does not exist. This was reported by a user of parca-agent, which is based on a fork of opentelemetry-ebpf-profiler: https://github.com/parca-dev/parca-agent/issues/2976 . However, I don't think the issue is unique to Parca, and exactly the same issue would happen in the stock OTel profiler.

The original reporter is running Unraid, but similar issues can happen in various other distributions that might have `CONFIG_FTRACE_SYSCALLS` or other relevant kernel configuration options disabled. For example, https://github.com/evilsocket/opensnitch/issues/774 documents similar brokenness in OpenSnitch when running on Liquorix kernels.

If I understand the OTel code correctly, we rely on the ability to attach syscall tracepoints in three places:
1. In `ProbeTracepoint`, called [here](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/main.go#L135-L137). This is just a check that syscall tracepoints work, so we can exit early in main if they are unavailable. It isn't actually functionally required.
2. In [`loadKernelCode`](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/tracer/systemconfig.go#L169-L169), called from [`loadTPBaseOffset`](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/tracer/tpbase.go#L49-L49), called from [`loadSystemConfig`](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/tracer/systemconfig.go#L241-L256). Note that this is only called if `parseBTF` fails. So it is unnecessarily in kernels where BTF-based offset discovery is available.
3. In [`loadKernelCode`](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/tracer/systemconfig.go#L169-L169), called from [`checkForMaccessPatch`](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/8014571/tracer/maccess.go#L30-L30). This is only necessary for the specific versions of the kernel that might have the [`copy_from_user_nofault` bug](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/mm/maccess.c?h=v6.1.36&id=2e7ad879e1b0256fb9e4703fd6cd2864d707dea7) which causes freezes.

So, there is a configuration that we could support, but don't now: if syscall tracepoints are unavailable, but BTF-based offset discovery is possible _and_ we're in a kernel version that is known not to be impacted by the `copy_from_user_nofault` bug, we will currently fail to run due to the lack of syscall tracepoints, even though logically the agent should be able to work.

Thus I propose that we don't require syscall tracepoints in the case when we're in such a configuration.

**Comment by umanwizard** [comment]
CC: @florianl

**Comment by rockdaboot** [comment]
If I understand correctly, it would boil down to something like
```go
if !tracer.BtfAvailable() || (kernelVersionCheck && tracer.HasProbeReadBug()) {
	if err = tracer.ProbeTracepoint(); err != nil {
		return failure("Failed to probe tracepoint: %v", err)
	}
}
```
`BtfAvailable()` straight forward and `HasProbeReadBug()` can be extracted from `tracer.go, L421ff`.

**Comment by fabled** [comment]
I think we could just remove `tracer.ProbeTracepoint` and its call completely. The subsequent attempts will fail anyway. I think that's a redundant check that just slow downs startup on current setup. @florianl  or was there some strange reason why this has to be done ahead of time?

**Comment by umanwizard** [comment]
I agree with @fabled 's suggested approach and this is what I proposed for Parca: https://github.com/parca-dev/parca-agent/pull/2990/files

There is one minor change that I would also suggest: making it clear in the error messages when `checkForMaccessPatch` that it's not necessarily the case that the user is running on an unpatched kernel, but rather that we don't know because we couldn't attach the tracepoint.

**Comment by umanwizard** [comment]
Implemented in #175
